### PR TITLE
added from_numpy

### PR DESCRIPTION
**Pull request by christine2515**
#1567

### CODE REVIEW

**Comment by simonetgordon** [comment]
Hi Christine,

Thanks for the contribution 😊

The code looks pretty good. It is currently failing the lint formatting check, so have a look at the 'Details' of this test to see what needs to be fixed. The other test failures are not related to your changes.

Simone

**Comment by simonetgordon** [comment]
Hi Christine,

No problem, I get an email notification every time you push a commit. Looks good, will merge 😊

Kind Regards, 

Simone

File: ivy/functional/ivy/creation.py [hunk 1]
```diff
@@ -690,6 +690,30 @@ def from_dlpack(x: Union[ivy.Array, ivy.NativeArray]) -> ivy.Array:
     return current_backend(x).from_dlpack(x)
 
 
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+def from_numpy(x: Union[ivy.Array, ivy.NativeArray]) -> ivy.Array:
```
Comment (simonetgordon) [inline ivy/functional/ivy/creation.py:696]: should only input numpy arrays

**Comment by simonetgordon** [comment]
HI Christine,

Actually there is an issue with the implementation, it currently only only works for np.arrays, yet you have ivy.Array + ivy.NativeArray in the function signature. It also outputs a tensor, not an ivy.Array. 

I'll get back to you clarifying if only np.arrays should be accepted. But for now, could you amend the function so that the function arguments and output type hint match the docstring description.

Kind Regards, 

Simone

**Comment by christine2515** [comment]
Hi Simone,

Thanks, just fixed it! 

I remember in the example from the Navigating the Code section of the Contributing guide, the backend API function had torch.Tensor as the output type hint while the Ivy API function signature had ivy.Array as the output type hint. Would that example also apply to the from_numpy function or is this an exception?

Thanks!
Christine

**Comment by simonetgordon** [comment]
Hi Christine,

It could be an exception. I will get back to you on this. Thanks again!

Simone

**Comment by christine2515** [comment]
Hi @simonetgordon,

Just wondering was there an update on this issue?

Thanks!
Christine

File: ivy/functional/backends/torch/creation.py [hunk 2]
```diff
@@ -373,6 +373,10 @@ def zeros_like(
     return torch.zeros_like(x, device=as_native_dev(device))
 
 
+def from_numpy(x):
```
Comment (simonetgordon) [inline ivy/functional/backends/torch/creation.py:376]: Add input and output type hints here: `from_numpy(x: np.ndarray) -> torch.Tensor`

**Comment by simonetgordon** [comment]
Hi Christine,

Sorry about the delay in getting back to you. Unfortunately, the `Add Creation Ops to PyTorch Frontend` issue is not live yet. Instructions will be added very soon, however, on the [website](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#open-tasks). I can't approve this PR until it is made sure it adheres to the guidelines.

You can either choose to wait until the instructions are published (perhaps 2-3 days) or you can do a [reformatting task](https://github.com/unifyai/ivy/issues?q=is%3Aopen+is%3Aissue+label%3A%22Function+Reformatting%22+label%3AToDo) if you don't want to wait.

Many apologies about this! 

Simone

**Comment by simonetgordon** [comment]
Hi Christine, do you plan to revisit this PR when the frontends task is live, or do you wish to close it?

**Comment by christine2515** [comment]
Hi Simone,

Thanks for the reminder, I totally forgot about it! I think I'll close it and work on one of the currently open tasks. How do I close the PR?

Thanks!
Christine

**Comment by simonetgordon** [comment]
Hi Christine,

No worries! I can close it 😊
